sw/device/tests/otbn_mem_scramble_test.c

```diff
@@ -22,19 +22,59 @@ typedef dif_result_t (*otbn_write_t)(const dif_otbn_t *otbn,
                                      uint32_t offset_bytes, const void *src,
                                      size_t len_bytes);
 
-/**
- * Number of distinct addresses to check in each IMEM and DMEM.
- */
-static const int kNumAddrs = 50;
-
-/**
- * Minimum number of expected integrity errors in IMEM and DMEM after
- * re-scrambling.
- *
- * Note that there is a non-zero chance for every word that the integrity bits
- * after re-scrambling are still valid.
- */
-static const int kNumIntgErrorsThreshold = kNumAddrs - 1;
+enum {
+  /**
+   * Number of distinct addresses to check in each IMEM and DMEM.
+   */
+  kNumAddrs = 50,
+
+  /**
+   * Minimum number of expected integrity errors in IMEM and DMEM after
+   * re-scrambling.
+   *
+   * Note that each word has 7 bits of ECC, so the probability that a word would
```

Comment:
For what it's worth, I just looked into implementation of `prim_secded_39_32_dec.sv`.
There are certainly some multi-bit errors that will trigger a single-bit error or a double-bit error.
However, everything that is not a valid code word will trigger one of the two errors.

Since we are working with 39-bit words and there are 2^32 valid code words, the probability that a randomly selected 39-bit word triggers an error is indeed 1 in 128.
If the test doesn't care which type of error is triggered, then the remaining calculations look correct to me.

Reply:
somewhat related, there was an old PR that described with valid code words and the "halo" of detection around it. See [here](https://github.com/lowRISC/opentitan/pull/3899#issuecomment-716799810).

I think what @vrozic is saying makes sense to me as well.